src/lib/util/cdrom.cpp

```diff
@@ -1625,6 +1658,60 @@ bool ecc_verify(const uint8_t *sector)
 	return true;
 }
 
+/**
+ * @fn  void edc_crctable_init(void)
+ *
+ * @brief   -------------------------------------------------
+ *           edc_crctable_init - calculate CRC checksums as per
+ *           ECMA 130 and store into lookup table EDC_crctable
+ *          -------------------------------------------------.
+ */
+
+void edc_crctable_init(void)
+{
+	for (int index = 0; index < EDC_CRCTABLE_SIZE; index++)
+	{
+		uint32_t r = reverse32(index);
+
+		for (int bit = 0; bit < 8; bit++)
+		{
+			if (r & 0x80000000L)
+				r = (r << 1) ^ EDC_POLYNOMIAL;
+			else
+				r <<= 1;
+		}
+
+		EDC_crctable[index] = reverse32(r);
+	}
+}
```

Comment:
There are mechanisms to allow this to be safely initialised without requiring a function call like this, e.g. function statics now have thread-safe initialisation (typically implemented using DCLP).

Reply:
This is not a class it can’t use DCLP.

Reply:
You can use a static instance of a small class inside a function to initialise. The compiler uses DCLP behind the scenes.

Reply:
How about this instead. Avoids the complexity of DCLP but achieves the same result.  resolved d55e9bff7ca8c66c7cbcde5b3a31ea36af9c752a

Reply:
That forces it to be initialised before main starts, even if it's never used. This adds up in something as big as MAME.

Reply:
It's computed at compile time and occupies 256 words. The code it removes would be bigger than that.